### PR TITLE
Add return authorization reason id to return items

### PR DESCRIPTION
**Pull request by plongyear**
Currently a return authorization reason can only be set on the entire return authorization. If we set return authorization reasons on return items, each item will be able to have a different reason.

### CODE REVIEW

**Comment by jordan-brough** [comment]
:+1:

File: core/app/models/spree/return_item.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@ class ReturnItem < ActiveRecord::Base
     belongs_to :reimbursement, inverse_of: :return_items
     belongs_to :preferred_reimbursement_type, class_name: 'Spree::ReimbursementType'
     belongs_to :override_reimbursement_type, class_name: 'Spree::ReimbursementType'
+    belongs_to :return_authorization_reason
```
Comment (athal7) [inline core/app/models/spree/return_item.rb]: you might need to specify the class_name here?

**Comment by athal7** [comment]
:+1:

File: core/app/models/spree/return_item.rb (hunk 1 above)
Comment (plongyear) [inline core/app/models/spree/return_item.rb]: It seems to work without the class name.

**Comment by plongyear** [comment]
There are a few more changes that need to be made to returns and I think it will be easier to review with all the changes together. I'm going to close this for now and re-open with the additional changes.

File: core/app/models/spree/return_item.rb (hunk 1 above)
Comment (athal7) [inline core/app/models/spree/return_item.rb]: cool!

**Pull request closed without merging**